cli/replicaset/roles.go

```diff
@@ -5,18 +5,57 @@ import (
 	"slices"
 )
 
-// ChangeRoleFunc is a function type for addition or removing a role.
+// RoleAction is a type that describes an action that will
+// happen with Roles. It can be "add" or "remove".
+type RoleAction uint
+
+const (
+	AddAction RoleAction = iota
+	RemoveAction
+)
+
+// ChangeRoleFunc is a function for adding/removing roles.
 type ChangeRoleFunc func([]string, string) ([]string, error)
 
-// AddRole is a function that implements addition of role.
+// RolesChangerAction implements changing (add or remove) of roles.
+type RolesChangerAction interface {
+	// ChangeRoleFunc is a function type for addition or removing a role.
+	Change([]string, string) ([]string, error)
+	// Action is a method that returns RoleAction.
+	Action() RoleAction
+}
+
+// RolesAdder is a struct that implements addition of role.
+type RolesAdder struct{}
+
+func (RolesAdder) Change(roles []string, r string) ([]string, error) {
+	return AddRole(roles, r)
+}
+
+func (RolesAdder) Action() RoleAction {
+	return AddAction
+}
+
+// RolesRemover is a struct that implements removing of role.
+type RolesRemover struct{}
+
+func (RolesRemover) Change(roles []string, r string) ([]string, error) {
+	return RemoveRole(roles, r)
+}
+
+func (RolesRemover) Action() RoleAction {
+	return RemoveAction
+}
```

Comment:
It seems like you could add unit-tests for the code. And for AddRole/RemoveRole (do we need it in the public API?) functions too.

Reply:
Added tests and deleted exportable `AddRole`, `RemoveRole` functions.